packages/blade/src/components/Text/Decisions/Decisions.md

```diff
@@ -0,0 +1,87 @@
+The text component is categorised into multiple parts
+
+- BaseText - Internal Component, not published to consumers
+- Title - Public Component
+- Heading - Public Component
+- Text - Public Component
+- Code - Public Component
+
+Here's how the hierarchy will looks like
+![Component Breakdown](./component-breakdown.png)
+
+The APIs for all the above component will look something like below
+
+- `BaseText`
+
+  This component is just meant to be used for internal purposes for Blade
+  Here's how the API will look like
+
+  ```jsx
+  <BaseText
+    color="surface.text | feedback.text | button.text"
+    fontFamily="fontfamily.name"
+    fontSize="fontSize.name"
+    fontWeight="fontWeight.name"
+    lineWeight="lineWeight"
+    letterSpacing="letterSpacing"
+    formatting="bold/italic/strike"
+    as="h1-h6/p/span"
```

Comment:
Will all exported components support `as`?

Reply:
nupe only base text which again will only be used internally

Reply:
For example this section in razorpay.com-

![image](https://user-images.githubusercontent.com/30949385/162694300-bdd63bd4-0999-4fdb-9b45-c05c2e46f462.png)

Here "Explore Razorpay Business Banking" is h2

Both, "Manage your company’s finances with Razorpay X Business Banking" and those labels like "Current Account", "Capital & Credits" they are all h3 even though they are styled completely differently.

Are we going to limit such usecases from design itself?

Reply:
yes that won't be a problem right since we have `Heading` tag to deal with these kind of use cases